app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -366,7 +380,21 @@ private void receiveSharedItems() {
             fragments = new ArrayList<>();
             for (UploadableFile uploadableFile : uploadableFiles) {
                 UploadMediaDetailFragment uploadMediaDetailFragment = new UploadMediaDetailFragment();
-                uploadMediaDetailFragment.setImageTobeUploaded(uploadableFile, place);
+                currLocation = locationManager.getLastLocation();
+                if (currLocation != null) {
+                    float locationDifference = getLocationDifference(currLocation, prevLocation);
+                    /* Remove location if app has access to it and is recording it but user
```

Comment:
This may require the same change too, sorry!

Reply:
This does not remove tags from EXIF. It just discards the location the app is recording. Considering the case where the user has unchecked the location EXIF tag from the Settings, I did not use the current location too in case the user taps "No".